conn.go

```diff
@@ -836,13 +863,37 @@ func (cn *conn) Exec(query string, args []driver.Value) (res driver.Result, err
 }
 
 func (cn *conn) send(m *writeBuf) {
+	defer func() {
```

Comment:
defer allocates.  At the very least don't make everyone pay a price for this feature.  But ideally don't defer at all.